cmake/External/nnpack.cmake

```diff
@@ -59,9 +59,12 @@ if(ANDROID OR IOS OR ${CMAKE_SYSTEM_NAME} STREQUAL "Linux" OR ${CMAKE_SYSTEM_NAM
   set(GOOGLETEST_SOURCE_DIR "${CAFFE2_THIRD_PARTY_ROOT}/googletest" CACHE STRING "Google Test source directory")
 
   if(NOT TARGET nnpack)
+    if(NOT USE_SYSTEM_PTHREADPOOL AND USE_INTERNAL_PTHREADPOOL_IMPL)
```

Comment:
When is this needed? For C2's NNPACK integration? If so can we use BUILD_CAFFE2_OPS?

Reply:
NNPACK is also used in PyTorch.  We want it to use the same thread pool as the rest of the code.  If we are building for C2, then NNPACK, QNNPACK, and PyTorch QNNPACK will all use C2's implementation.  Otherwise everything will use the third-party version.  So it's all this, or all that and never a mix.

The reason I am not using BUILD_CAFFE2_OPS here is because I want to centralize the logic in a single place.  That central place decides whether to USE_INTERNAL_PTHREADPOOL_IMPL or not.  If that logic has come to the conclusion to use C2's implementation, either because BUILD_CAFFE2_OPS is ON, or we are on Windows, or whatever other condition, then that sets the USE_INTERNAL_PTHREADPOOL_IMPL flag and that way at least we wouldn't have to duplicate that error prone logic everywhere.